### PR TITLE
Fix up project search scenarios when no option is provided

### LINKED ISSUE #42794: user-jwts prints error message on `path1`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When executing `dotnet user-jwts create` in a folder location the error printed is:
```
Value cannot be null. (Parameter 'path1')
```
It does not tell me what value I'm looking for nor is `path1` helpful enough for me to resolve.

### Expected Behavior

Specific missing parameter name that helps me resolve

### Steps To Reproduce

Go to a folder root where you have a solution (but the project is in a folder underneath) and execute `dotnet user-jwts create`

### Exceptions (if any)

_No response_

### .NET Version

7.0.100-preview.6.22352.1

### Anything else?

.NET SDK:
 Version:   7.0.100-preview.6.22352.1
 Commit:    492644e08e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100-preview.6.22352.1\

Host:
  Version:      7.0.0-preview.6.22324.4
  Architecture: x64
  Commit:       d3fa592f6d

.NET SDKs installed:
  3.1.421 [C:\Program Files\dotnet\sdk]
  5.0.408 [C:\Program Files\dotnet\sdk]
  5.0.410 [C:\Program Files\dotnet\sdk]
  6.0.100-preview.3.21202.5 [C:\Program Files\dotnet\sdk]
  6.0.100-preview.4.21226.14 [C:\Program Files\dotnet\sdk]
  6.0.100-preview.4.21255.9 [C:\Program Files\dotnet\sdk]
  6.0.202 [C:\Program Files\dotnet\sdk]
  6.0.203 [C:\Program Files\dotnet\sdk]
  6.0.302 [C:\Program Files\dotnet\sdk]
  6.0.400-preview.22330.6 [C:\Program Files\dotnet\sdk]
  7.0.100-preview.6.22352.1 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.27 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0-preview.6.22330.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.27 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-preview.6.22324.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.26 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.27 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0-preview.6.22351.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

### PR DESCRIPTION
**Pull request by captainsafia**
Closes https://github.com/dotnet/aspnetcore/issues/42794.

### CODE REVIEW

File: src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs [hunk 1]
```diff
@@ -480,4 +480,88 @@ public void Key_CanPrintAndReset_BySchemeAndIssuer()
...
+    }
+
+    [Fact]
+    public void Create_CanHandleNoProjectOptionProvided_WithNoProjects()
+    {
+        var projectPath = _fixture.CreateProject();
+        var projectParent = Path.Combine(projectPath, "..");
+
+        Directory.Delete(projectParent, true);
+        Directory.CreateDirectory(projectParent);
```
Comment (BrennanConroy) [inline src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs:505]: This is an expensive way to create an empty folder

File: src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs [hunk 2]
```diff
@@ -32,7 +32,7 @@ public static string GetProject(string projectPath = null)
             return projectPath;
         }
 
-        var csprojFiles = Directory.EnumerateFileSystemEntries(Directory.GetCurrentDirectory(), "*.*proj", SearchOption.TopDirectoryOnly)
+        var csprojFiles = Directory.EnumerateFileSystemEntries(Directory.GetCurrentDirectory(), "*.*proj", SearchOption.AllDirectories)
```
Comment (BrennanConroy) [inline src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs:35]: Is this guaranteed to find the top most `*proj`? Like if my project was `b.csproj` but had a folder `a/` with an `a.csproj`, that wouldn't be used instead?

File: src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs [hunk 3]
```diff
@@ -112,7 +112,10 @@ public static byte[] CreateSigningKeyMaterial(string userSecretsId, string schem
 
     public static List<string> GetAudienceCandidatesFromLaunchSettings(string project)
     {
-        ArgumentException.ThrowIfNullOrEmpty(nameof(project));
+        if (project is null)
```
Comment (BrennanConroy) [inline src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs:115]: `string.IsNullOrEmpty`?

**Review by BrennanConroy — APPROVED** [review]

File: src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs (hunk 1 above)
Comment (captainsafia) [reply src/Tools/dotnet-user-jwts/test/UserJwtsTests.cs:505]: Fair point!

File: src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs (hunk 2 above)
Comment (captainsafia) [reply src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs:35]: The current behavior will throw an error for the multiple project scenario as it only supports looking for a single project file within a sub-directory. We can do something like use the first project file it finds but I'm not sure that's the best option. It feels more sensible to force the user to be explicit if we cannot find something.

Comment (DamianEdwards) [reply src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs:35]: We should likely just match whatever the behavior of `user-secrets` is yes?

Comment (captainsafia) [reply src/Tools/dotnet-user-jwts/src/Helpers/DevJwtCliHelpers.cs:35]: `user-secrets` only searches the current directory if we wanted to stick to that.

**Pull request merged**